common/server_room.cpp

```diff
@@ -232,6 +234,16 @@ void Server_Room::say(const QString &userName, const QString &s, bool sendToIsl)
     event.set_name(userName.toStdString());
     event.set_message(s.toStdString());
     sendRoomEvent(prepareRoomEvent(event), sendToIsl);
+
+    ServerInfo_ChatMessage chatMessage;
+    chatMessage.set_time(QDateTime::currentDateTime().toString().toStdString());
+    chatMessage.set_sender_name(userName.toStdString());
+    chatMessage.set_message(s.simplified().toStdString());
+
+    if (chatHistory.size() > 100)
```

Comment:
I thought about this, the other area I thought it may fit was when the room is defined.  Thoughts to either?  It doesn't matter to me tbh.